src/backend/distributed/planner/distributed_planner.c

```diff
@@ -881,6 +896,53 @@ TryCreateDistributedPlannedStmt(PlannedStmt *localPlan,
 }
 
 
+/*
+ * GetRouterPlanType checks the parse tree to return appropriate plan type.
+ */
+static RouterPlanType
+GetRouterPlanType(Query *query, Query *originalQuery, bool hasUnresolvedParams)
+{
+	if (IsModifyCommand(originalQuery))
+	{
+		Oid targetRelationId = ModifyQueryResultRelationId(query);
+
+		EnsureModificationsCanRunOnRelation(targetRelationId);
+		EnsurePartitionTableNotReplicated(targetRelationId);
+	}
+	else
+	{
+		return SELECT_QUERY;
```

Comment:
minor: would be slightly nicer to have an
```
if (!IsModifyCommand(originalQuery))
{
    return SELECT QUERY;
}

Oid targetRelationId = ModifyQueryResultRelationId(query);
...
```
to make a clear separation between reads and writes